src/ConnectedMode.UnitTests/Binding/ImportBeforeInstallTriggerTests.cs

```diff
@@ -34,56 +33,130 @@ public class ImportBeforeInstallTriggerTests
+            var threadHandling = new Mock<IThreadHandling>();
+
+            _ = CreateTestSubject(activeSolutionTracker.Object, threadHandling: threadHandling.Object);
 
+            threadHandling.Verify(x => x.SwitchToBackgroundThread(), Times.Once);
+        }
+
+        [TestMethod]
+        public void InvokeBindingChanged_Standalone_ImportBeforeFileGeneratorIsNotCalled()
+        {
+            var activeSolutionTracker = new Mock<IActiveSolutionBoundTracker>();
+            var importBeforeFileGenerator = new Mock<IImportBeforeFileGenerator>();
+
+            _ = CreateTestSubject(activeSolutionTracker.Object, importBeforeFileGenerator.Object);
+
+            activeSolutionTracker.Raise(x => x.PreSolutionBindingChanged += null, new ActiveSolutionBindingEventArgs(BindingConfiguration.Standalone));
             importBeforeFileGenerator.Verify(x => x.WriteTargetsFileToDiskIfNotExists(), Times.Never);
         }
 
         [TestMethod]
         [DataRow(SonarLintMode.Connected)]
         [DataRow(SonarLintMode.LegacyConnected)]
-        public void TriggerUpdate_ConnectedMode_ImportBeforeFileGeneratorIsCalled(SonarLintMode mode)
+        public void InvokeBindingChanged_ConnectedMode_ImportBeforeFileGeneratorIsCalled(SonarLintMode mode)
         {
-            var configProvider = SetUpConfigProvider(mode);
+            var activeSolutionTracker = CreateActiveSolutionBoundTrackerWihtBindingConfig(SonarLintMode.Standalone);
             var importBeforeFileGenerator = new Mock<IImportBeforeFileGenerator>();
 
-            var testSubject = CreateTestSubject(configProvider, importBeforeFileGenerator.Object);
-            testSubject.TriggerUpdate().Forget();
+            _ = CreateTestSubject(activeSolutionTracker.Object, importBeforeFileGenerator.Object);
+
+            importBeforeFileGenerator.Verify(x => x.WriteTargetsFileToDiskIfNotExists(), Times.Never);
 
+            var newBindingConfig = CreateBindingConfiguration(mode);
+            activeSolutionTracker.Setup(x => x.CurrentConfiguration).Returns(newBindingConfig);
+
+            activeSolutionTracker.Raise(x => x.PreSolutionBindingChanged += null, new ActiveSolutionBindingEventArgs(newBindingConfig));
             importBeforeFileGenerator.Verify(x => x.WriteTargetsFileToDiskIfNotExists(), Times.Once);
         }
 
-        private IConfigurationProvider SetUpConfigProvider(SonarLintMode mode)
+        [TestMethod]
+        public void InvokeBindingUpdated_ImportBeforeFileGeneratorIsCalled()
         {
-            var configProvider = new Mock<IConfigurationProvider>();
-            var bindingConfig = new BindingConfiguration(
-                                    new BoundSonarQubeProject(new Uri("http://localhost"), "test", ""),
-                                    mode, "");
+            var activeSolutionTracker = CreateActiveSolutionBoundTrackerWihtBindingConfig(SonarLintMode.Standalone);
```

Comment:
I think you can extract this and the Changed test into one method that receives RaiseBindingEvent method. But it's minor, your call.